mcan/src/message/tx.rs

```diff
@@ -108,30 +108,40 @@ impl<'a> MessageBuilder<'a> {
     pub fn build<const N: usize>(self) -> Result<Message<N>, TooMuchData> {
         let mut data = [0; N];
 
+        let mut copy_payload = |d: &[u8]| {
+            if d.len() > N {
+                return Err(TooMuchData);
+            }
+            data[..d.len()].copy_from_slice(d);
+            Ok(())
+        };
```

Comment:
Do you prefer a closure here over a function?

```rust
    fn copy_payload<const N: usize>(dest: &mut [u8; N], src: &[u8]) -> Result<(), TooMuchData> {
        if src.len() > N {
            return Err(TooMuchData);
        }
        dest[..src.len()].copy_from_slice(src);
        Ok(())
    };
```

Reply:
Closure does not require explicit signature and is still better than a macro. So :woman_shrugging: